Generator.md

```diff
@@ -0,0 +1,356 @@
+    {
+        public static IServiceCollection RegisterFromAgodaExample(this IServiceCollection serviceCollection)
+        {
+            return Register(serviceCollection);
+        }
+
+        internal static IServiceCollection Register(this IServiceCollection serviceCollection)
+        {
+ serviceCollection.AddSingleton<IService, Service>;
+            return serviceCollection;
+        }
+    }
+}
+```
+Factory options are available for registration with the attributes, as seen below, the factory needs a "Factory" method and Factory class should implement the interface called `IImplementationFactory<of your targer type>`
+
+Let's examine one example.
+
+```csharp
+using Agoda.IoC.Generator.Abstractions;
+
+namespace Agoda.Example;
+
+public interface ICustomerService { }
+
+[RegisterSingleton(Factory = typeof(CustomerServiceFactory))]
+public class CustomerService : ICustomerService
+{
+}
+
+// Factory class
+public class CustomerServiceFactory : IImplementationFactory<ICustomerService>
+{
+    public ICustomerService Factory(IServiceProvider serviceProvider)
+    {
+        // more code here
+        // serviceProvider.GetService(typeof(AccoutService));
+        return new CustomerService();
+    }
+}
+```
+
+Project analyzers contain the complete source code.
+
+```csharp
+// <auto-generated>
+//     Automatically generated by Agoda.IoC.Generator.
+//     Changes made to this file may be lost and may cause undesirable behavior.
+// </auto-generated>
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Agoda.Example;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class AgodaIoCGeneratorInjectServiceCollectionExtension
+    {
+        public static IServiceCollection RegisterFromAgodaExample(this IServiceCollection serviceCollection)
+        {
+            return Register(serviceCollection);
+        }
+
+        internal static IServiceCollection Register(this IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddSingleton<IService, Service>();
+            serviceCollection.AddSingleton(sp => new CustomerServiceFactory().Factory(sp));
+            return serviceCollection;
+        }
+    }
+}
+```
+
+For services with multiple interfaces the interface can be explicitly declared like below
+
+``` csharp
+// This class implements 2 interfaces, but we explicitly tell it to register only 1.
+[RegisterTransient(For = typeof(IExplicitlyRegisteredInterface))]
+public class ServiceWithExplicitInterfaceRegistration : IExplicitlyRegisteredInterface, IInterfaceThatShouldNotGetRegistered {}
+```
+It can also be used to register multiple instances
+
+``` csharp
+[RegisterSingleton(For = typeof(IMultipleAttributes1))]
+[RegisterSingleton(For = typeof(IMultipleAttributes2))]
+public class MultipleAttributes : IMultipleAttributes1, IMultipleAttributes2 {}
+```
+
+Additionally, it may be used to assign an attribute order and register many instances as a collection.
```

Comment:
Is this existing functionality or new?

I think if its existing we should use the same docs to describe both.

If it's new we should update the original library to do the same then again only have one doc that describes functionality

Reply:
This feature already exist so i will have to add on both